File: dynamic_programming/python/pascal_triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        list = []
        list.append([1])
        for i in range(1,numRows):
            temp = []
            size = len(list[len(list)-1])
            temp.append(1)
            if size>1:
                for j in range(size-1):
                    temp.append(list[size-1][j]+list[size-1][j+1])
            temp.append(1)
            list.append(temp)
        return list

File: dynamic_programming/c++/pascal_triangle.cpp
#include <iostream>
#include <vector>

class Solution {
public:
    vector<vector<int>> generate(int rows){
        vector<vector<int>> res = {};
        res.push_back({1});
        for(int i=1;i<rows;i++){
            vector<int> temp = {};
            temp.push_back(1);
            int size = res[res.size()-1].size();
            if(size>1){
                for(int j=0;j<size-1;j++){
                    temp.push_back(res[size-1][j]+res[size-1][j+1]);
                }
            }
            temp.push_back(1);
            res.push_back(temp);
        }
        return res;
    }
};
